sms-checker/checker.py

```diff
@@ -0,0 +1,44 @@
+'''This script will check the database to see whether any subscribers require notification'''
+import logging
+from os import environ as ENV
+from dotenv import load_dotenv
+from psycopg2 import connect
+
+def get_connection():
+    '''Returns a psycopg2 connection'''
+    return connect(f"""dbname={ENV["DB_NAME"]} user={ENV["DB_USER"]}
+                 host={ENV["DB_HOST"]} password={ENV["DB_PASSWORD"]} port={ENV["DB_PORT"]}""")
+
+def get_subscribers() ->list[dict]:
+    '''Returns a list of dictionaries containing subscribers,
+      bodies, time and notification service'''
+    query = '''SELECT s.subscriber_username,s.subscriber_phone,
+    s.subscriber_email,b.body_name,ba.at as body_time
+    FROM subscriber_county_assignment as sca
+    JOIN subscriber as s USING (subscriber_id)
+    JOIN county as c USING (county_id)
+    JOIN forecast as f USING (county_id)
+    JOIN region as r USING (region_id)
+    JOIN body_assignment as ba USING (region_id)
+    JOIN body as b USING (body_id)
+    WHERE f.visibility_m > 500 AND
+    f.cloud_coverage_percent < 25 AND
+    ba.at <= CURRENT_TIMESTAMP + INTERVAL '4 hours'
+    AND ba.at > CURRENT_TIMESTAMP '''
```

Comment:
Good query that should ensure we are only giving people notifications when it is likely they will be able to see the night sky.